static/js/containers/ProfileFormContainer.js

```diff
@@ -112,15 +113,19 @@ class ProfileFormContainer extends React.Component {
     dispatch(startProfileEdit(username));
   };
 
-  saveProfile(isEdit: boolean, validator: Validator|UIValidator, profile: Profile, ui: UIState) {
+  saveProfile(isEdit: boolean, validator: Validator|UIValidator|null, profile: Profile, ui: UIState) {
```

Comment:
The `validator` is a function which validates part or all of the profile. `saveProfile` runs the validator and, if there are no errors, it issues the `PATCH` request to save the updated profile.

The reason for the `validator` is that we do not have any restriction on the backend for requiring fields. We need to do this because we fill the profile out in steps, so we want to be able to save some fields, take the user to the next page, and save some more. So we don't always want to validate the whole profile, sometimes we want to validate just the section of interest.

I do think here there should be a `validator` which does not let the user save until they've entered something for 'about me'. That would be in line with the pattern we use everywhere else (that passing a `validator` to `saveProfile` is required).